new_work18/js/script.js

```diff
@@ -0,0 +1,164 @@
+    displayDiv.after(dotDiv);
+
+    //prev/next button
+    const prevButton = document.createElement('button');
+    prevButton.type = 'button';
+    prevButton.textContent = '◀';
+    prevButton.addEventListener('click', function(){
+        showImage(getActive() - 1);
+    }, false);
+
+    const nextButton = document.createElement('button');
+    nextButton.type = 'button';
+    nextButton.textContent = '▶';
+    nextButton.addEventListener('click', function(){
+        showImage(getActive() + 1);
+    }, false);
+
+    sliderUl.before(prevButton);
+    sliderUl.after(nextButton);
+
+    //current
+    const currentDiv = document.createElement('div');
+    currentDiv.classList.add('current')
+    dotDiv.after(currentDiv);
+
+    //slideListsの設定
+    slideLists = [...document.querySelectorAll( '.slider__list' )];
+
+    showImage(0);
+}
+
+function updateControl() {
+    //prev/next button
+    const prevButton = sliderUl.previousElementSibling;
+    const nextButton = sliderUl.nextElementSibling;
+    if(getActive() === 0){
+        prevButton.disabled = true;
+    } else {
+        prevButton.disabled = false;
+    }
+
+    if(getActive() === slideLists.length - 1){
+        nextButton.disabled = true;
+    } else {
+        nextButton.disabled = false;
+    }
+
+    //current
+    const currentDiv = document.querySelector('.current');
+    currentDiv.textContent = `${getActive() + 1}/${slideLists.length}`;
+
+    //dot button
+    const dotDiv = document.querySelector('.dot');
+    [...dotDiv.children].forEach(item => item.disabled = false);
+    dotDiv.children[getActive()].disabled = true;
+}
+
+function setTimer() {
+    clearTimeout(timer);
+    timer = setTimeout(nextImage, 3000);
+}
+
+function showImage(value) {
+    const activeLi = document.querySelector('.is-active');
+    if(activeLi) {
+        activeLi.classList.remove('is-active');
+    }
+    slideLists[value].classList.add('is-active');
+    updateControl();
+    setTimer();
+}
+
+function nextImage() {
+    if(getActive() < slideLists.length - 1 ) {
+        showImage(getActive() + 1);
+    } else {
+        showImage(0);
+    }
+}
+
+function getJson() {
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            const response = fetch('https://jsondata.okiba.me/v1/json/grhYz210215052753');
+            resolve(response);
+        }, 3000);
+    });
+};
+
+async function getJsonimage() {
+    const loading = document.createElement('img');
+    loading.src = "img/loading-circle.gif";
+    sliderUl.appendChild(loading);
+
+    try {
+        const response = await getJson();
+        const resJson = await response.json();
+        return resJson.image;
+    } catch (err) {
+        sliderUl.innerHTML = 'ただいまサーバー側で通信がぶっ壊れています';
+        throw err;
+    } finally {
+        loading.remove();
+    }
+
+};
+
+window.onload = async function(){
+    const resImage = await getJsonimage()
+    initSlider(resImage);
+}
```

Comment:
getJsonImageなので
それ以外の仕事はやらない方がいい気がしました

```suggestion
async function getJsonimage() {
    try {
        const response = await getJson();
        const resJson = await response.json();
        return resJson.image;
    } catch (err) {
        sliderUl.innerHTML = 'ただいまサーバー側で通信がぶっ壊れています';
        throw err;
    }
};

window.onload = async function(){
       const loading = document.createElement('img');
    loading.src = "img/loading-circle.gif";
    sliderUl.appendChild(loading);
    const resImage = await getJsonimage()
            loading.remove();
    initSlider(resImage);
}
```

Reply:
修正しました。　https://github.com/matsuwork/markup-work/pull/7/commits/0717d18d307c6c438a13eaee3c4d21dbc918cf12